general/development/process/peer-review/accessibility-checklist.md

```diff
@@ -0,0 +1,195 @@
+
+### Use of tabindex
+
+Tabindex > 0 is wrong. Using any tabindex value other than 0 (for focusable elements) or -1 (for non-focusable elements) is an anti-pattern. It can cause unexpected behaviour for keyboard users.
+
+### Zoom test
+
+Test the page in 200% and 400% zoom. If the page is not usable or elements are overlapped, the changes must be fixed before continuing.
+
+### Icons and images
+
+If the patch uses icons or images, check if they has a proper `alt` text.
+
+Most icons should have an empty alt text for icons, particularly in interactive containers such as buttons or links. This will assist mouse users, particularly when hovering over the icon button/link, which will display a tooltip without hovering over the icon itself.
+
+Decorative images should have an empty alt text (alt=""). In contrast, icons should have a descriptive alt text unless the alt text will be redundant with the surrounding elements.
+
+### Buttons
+
+Buttons with no comprehensive text should have an `aria-label` or `aria-labelledby` attribute.
+
+However, buttons with only icons should:
+
+- The icon itself should have an empty `alt` attribute so assistive technologies will ignore it.
+- Extra text must added to the button using a `span` with the `sr-only` class. This text will be only visible for assistive technologies.
+
+Here is an example of a `button` with a descriptive `sr-only` text:
+
+```html
+<button type="button" class="btn btn-primary">
+  <img src="./pix/add_user.png" alt="">
+  <span class="sr-only">Add a new user</span>
+</button>
+```
+
+And this another one using fontawesome icons:
+
+```html
+<button type="button" class="btn btn-primary">
+  <i class="icon fa-user-plus fa-fw " aria-hidden="true"></i>
+  <span class="sr-only">Add a new user</span>
+</button>
+```
+
+### Visible focus
+
+All focusable elements should have a visible focus indicator that should be **different** from the default browser focus indicator. The focus indicator should be visible in all states (normal, hover, focus, active, disabled).
+
+By default, Moodle override the focus indicator of most HTML elements to ensure accessibility. However, in some cases it is necessary to add extra CSS rules to ensure the focus indicator is visible and has enough contrast. For those cases, Moodle provides some CSS helpers:
+
+- Adding `aalink` to a link (or any other element acting as a link) will ensure the focus indicator is visible.
+- The CSS class `focus-visible` should be used to ensure the focus indicator is only visible when the element is focused using the keyboard.
+- In action menus or action menu sub panels, you can use `dropdown-item-outline` to ensure the focus indicator is visible on all the item and not only in the link element within.
+
+### Page headers and title
+
+Each new page should have a specific and descriptive page title. For example, use "Settings in forum ACTIVITYNAME" instead of just "Activity settings".
```

Comment:
```suggestion
Each new page should have a specific and descriptive page title.

:::info Example

Use "Settings in forum ACTIVITYNAME" instead of just "Activity settings".

:::
```